x-pack/legacy/plugins/lens/public/persistence/saved_object_store.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectAttributes } from 'target/types/server';
+
+export interface Document {
+  id?: string;
+  type?: string;
+  visualizationType: string | null;
+  datasourceType: string | null;
+  title: string;
+  state: {
+    datasource: unknown;
+    visualization: unknown;
+  };
+}
+
+interface SavedObectStore {
+  create: (type: string, object: SavedObjectAttributes) => Promise<{ id: string }>;
+  update: (type: string, id: string, object: SavedObjectAttributes) => Promise<{ id: string }>;
+  get: (
+    type: string,
+    id: string
+  ) => Promise<{ id: string; type: string; attributes: SavedObjectAttributes }>;
+}
+
+const DOC_TYPE = 'lens';
```

Comment:
Could you use the existing constant from `lens/common`? If we change the name, should this change as well?

Reply:
Tricky. We'd need a migration if the type itself changes. So, I'm not sure. The two should be the same, but they do signify different meanings.